### PR TITLE
feat: add js translations job

### PR DESCRIPTION
**Pull request by brian-smith-tcril**
this has been tested and is working on [my fork](https://github.com/brian-smith-tcril/openedx-translations). the version i have on my `main` branch on the fork is slightly different (uses `GITHUB_TOKEN` instead of `EDX_TRANSIFEX_BOT_GITHUB_TOKEN`, and pulls from my forks of each repo), but this PR does not include those changes.

### CODE REVIEW

File: .github/workflows/extract-translation-source-files.yml [hunk 1]
```diff
@@ -105,9 +107,78 @@ jobs:
...
+        # * frontend-enterprise?
+        # * frontend-app-learner-portal-enterprise
+        # * frontend-build?
+        # * frontend-app-learner-portal-programs
+        # * frontend-component-cookie-policy-banner
+        # * frontend-app-programs-dashboard
+        # studio-frontend has a different file structure
+        # repos with errors running extract_translations
+        # * frontend-template-application
+        repo: [frontend-app-payment, frontend-app-discussions, frontend-app-enterprise-public-catalog, frontend-app-learning, frontend-app-authn, frontend-app-course-authoring, frontend-app-learner-record, frontend-app-support-tools, frontend-component-header, frontend-component-footer, frontend-app-library-authoring, frontend-app-account, frontend-app-profile, frontend-app-ecommerce, frontend-app-gradebook, frontend-app-program-console]
```
Comment (Carlos-Muniz) [inline .github/workflows/extract-translation-source-files.yml:128]:
```suggestion
        repo: 
          - frontend-app-payment
          - frontend-app-discussions
          - frontend-app-enterprise-public-catalog
          - frontend-app-learning
          - frontend-app-authn
          - frontend-app-course-authoring
          - frontend-app-learner-record
          - frontend-app-support-tools
          - frontend-component-header
          - frontend-component-footer
          - frontend-app-library-authoring
          - frontend-app-account
          - frontend-app-profile
          - frontend-app-ecommerce, 
          - frontend-app-gradebook, 
          - frontend-app-program-console
```

File: .github/workflows/extract-translation-source-files.yml [hunk 2]
```diff
@@ -105,9 +107,78 @@ jobs:
...
     runs-on: ubuntu-latest
     needs: [setup-branch, python-translations]
+    steps:
+      # Clones the openedx-translations repo
+      - name: clone brian-smith-tcril/openedx-translations
```
Comment (Carlos-Muniz) [inline .github/workflows/extract-translation-source-files.yml:133]:
```suggestion
      - name: clone openedx/openedx-translations
```

File: .github/workflows/extract-translation-source-files.yml [hunk 3]
```diff
@@ -105,9 +107,78 @@ jobs:
...
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ needs.setup-branch.outputs.branch }}
+
+      # Clones the  repository
+      - name: clone brian-smith-tcril/${{ matrix.repo }}
```
Comment (Carlos-Muniz) [inline .github/workflows/extract-translation-source-files.yml:139]:
```suggestion
      - name: clone openedx/${{ matrix.repo }}
```

File: .github/workflows/extract-translation-source-files.yml [hunk 4]
```diff
@@ -105,9 +107,78 @@ jobs:
...
+        uses: actions/checkout@v3
+        with:
+          repository: brian-smith-tcril/${{ matrix.repo }}
```
Comment (Carlos-Muniz) [inline .github/workflows/extract-translation-source-files.yml:142]:
```suggestion
          repository: openedx/${{ matrix.repo }}
```

File: .github/workflows/extract-translation-source-files.yml [hunk 5]
```diff
@@ -105,9 +107,78 @@ jobs:
...
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+
+      # Extracts the translation source files
+      - name: extract translation source files
+        run: |
+          cd ${{ matrix.repo }}
+          make extract_translations
+      # git adds only the translation source files, found in conf/locale/en
```
Comment (Carlos-Muniz) [inline .github/workflows/extract-translation-source-files.yml:156]: Could you update this comment to reflect the translation file structure in MFEs

File: .github/workflows/extract-translation-source-files.yml [hunk 6]
```diff
@@ -105,9 +107,78 @@ jobs:
...
+          # Check the git status of the translation source files
+          echo "GIT_STATUS=$(git status ${{ matrix.repo }}/src/i18n/transifex_input.json -s | wc -l)" >> $GITHUB_ENV
+      # Attempts to commit the translation source files if there is a difference
+      - name: git commit the translation source files
+        if: "${{ env.GIT_STATUS > 0 }}"
+        run: |
+          # commit the changes
+          git commit -m "chore: add extracted translation source files from ${{ matrix.repo }}"
+          # push changes to branch
+          git pull
```
Comment (Carlos-Muniz) [inline .github/workflows/extract-translation-source-files.yml:176]: Since everything is running sequentially, is the `git pull` necessary here?

**Review by Carlos-Muniz — CHANGES_REQUESTED** [review]
Looks pretty good. Some changes are needed:

File: .github/workflows/extract-translation-source-files.yml (hunk 6 above)
Comment (brian-smith-tcril) [reply .github/workflows/extract-translation-source-files.yml:176]: tested [here](https://github.com/brian-smith-tcril/openedx-translations/actions/runs/3396813612/jobs/5648306115) and it seems to work without the pull, updating it now

File: .github/workflows/extract-translation-source-files.yml [hunk 7]
```diff
@@ -105,9 +108,93 @@ jobs:
...
+          - frontend-app-course-authoring
+          - frontend-app-learner-record
+          - frontend-app-support-tools
+          - frontend-component-header
+          - frontend-component-footer
+          - frontend-app-library-authoring
+          - frontend-app-account
+          - frontend-app-profile
+          - frontend-app-ecommerce, 
+          - frontend-app-gradebook,
```
Comment (Carlos-Muniz) [inline .github/workflows/extract-translation-source-files.yml:144]:
```suggestion
          - frontend-app-ecommerce
          - frontend-app-gradebook
```

**Review by Carlos-Muniz — CHANGES_REQUESTED** [review]

**Review by Carlos-Muniz — APPROVED** [review]

File: .github/workflows/extract-translation-source-files.yml [hunk 8]
```diff
@@ -105,9 +108,93 @@ jobs:
...
+          - frontend-app-learner-record
+          - frontend-app-support-tools
+          - frontend-component-header
+          - frontend-component-footer
+          - frontend-app-library-authoring
+          - frontend-app-account
+          - frontend-app-profile
+          - frontend-app-ecommerce
+          - frontend-app-gradebook
+          - frontend-app-program-console
```
Comment (arbrandes) [inline .github/workflows/extract-translation-source-files.yml:145]: Mind adding frontend-app-ora-grading as well?  Reason: it's being explicitly considered [for Olive](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3571056752/Olive+MFE+Candidate+Requirements+Matrix).

Comment (brian-smith-tcril) [reply .github/workflows/extract-translation-source-files.yml:145]: ah, it's on the `edx` org, not the `openedx` org, that's why i missed it https://github.com/edx/frontend-app-ora-grading

**Pull request merged**